server/server_test.go

```diff
@@ -0,0 +1,271 @@
+package server
```

Comment:
The old zenodb_test simulated clusters but didn't actually use the real RPC logic. This new test actually uses the real RPC logic to make sure that code-path is exercised.